docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md

```diff
@@ -0,0 +1,82 @@
+canonicalLink: https://www.taniarascia.com/migrating-from-wordpress-to-gatsby/
+---
+
+On September 24th, 2015, I wrote my [first article](https://www.taniarascia.com/getting-started-with-git/) on [taniarascia.com](https://www.taniarascia.com), which was a custom, self-hosted WordPress theme. I discovered Git, I discovered WordPress, and I [made 1,039 commits](https://github.com/taniarascia/oblate) to the theme, in which I obsessively [designed](https://www.taniarascia.com/version-2-0-website-redesign-863-commits-later/) and [redesigned](https://www.taniarascia.com/website-redesign-version-4-0/) the site.
+
+I've had quite a bit of experience with WordPress. Once I learned how to create a WordPress blog, I wrote [Developing a WordPress Theme from Scratch](https://www.taniarascia.com/developing-a-wordpress-theme-from-scratch/), a post that has had millions of hits, hundreds of comments, made me plenty of friends, and launched my blogging career. I also worked as a WordPress developer for two years. For the past four years, the blog had been running on WordPress. It was as fast, as custom, and as free of plugins as I could possibly make it, but WordPress is cumbersome, and there's only so fast and pleasant to use you can make it.
+
+After 10 days of obsessively working non-stop, I've finally migrated my site over to a [Gatsby](/)! You may notice the site is a little faster now:
+
+![](./speed.png)
+
+## Why Gatsby?
+
+There are [a lot of static site generators](https://www.staticgen.com/) to choose from. [Jekyll](https://jekyllrb.com/), [Hugo](https://gohugo.io/), [Next](https://nextjs.org/), and [Hexo](https://hexo.io/) are some of the big ones, and I've heard of and looked into some interesting up-and-coming SSGs like [Eleventy](https://www.11ty.io/) as well. At first, I thought I'd just want something that outputs straight HTML, and that a heavy JavaScript app couldn't possibly be better than simple HTML and CSS.
+
+However, I realized that a static site generator like Gatsby utilizes the power of code/data splitting, pre-loading, pre-caching, image optimization, and all sorts of performance enhancements that would be difficult or impossible to do with straight HTML.
+
+Since I primarily write JavaScript these days, I wanted an SSG that runs on Node.js, and if it uses React, even better. I tested out a few sites that run on Gatsby and yeah - they were fast. Blazingly fast.
+
+### A few things I really like about Gatsby
+
+- **No page reloads** - this site is now a SPA (single page app), and clicking on any internal page from within the website doesn't need to load a completely new resource
+- **Image optimization** - all the images are automatically stripped of metadata, optimized, resized, lazy-loaded, and compressed
+- **Pre-fetch resources** - Gatsby detects what links are available on a given page and loads that data into the cache
+- **Bundling and minification** - code is minified, bundled, and served
+- **No server side rendering**
```

Comment:
Gatsby does server rendering though...?

Reply:
It prerenders to build and deliver static assets. @DSchau 	how do you normally like to make this distinction.

Reply:
Gatsby does server-side rendering at build time.

It doesn't _require_ a Node.js server (e.g. like Next.js does), which saves a lot of 🤑.

```suggestion
- **Server-side rendered, at build time** Gatsby builds optimized static assets, which can be hosted anywhere!
```